### PR TITLE
Issues 24 & 25

### LINKED ISSUE #24: Change remaining time
set the remaining time to the next Azan not iqamah

### LINKED ISSUE #25: read yearly prayer times

### PR DESCRIPTION
**Pull request by gemy21ce**
- set the time left to next Azan instead of next Iqamah
- reading next month time table if less than 10 days left in the current month data

### CODE REVIEW

**Review by OmarHatem28 — APPROVED** [review]

File: test/data/network/prayer_times_remote_source_test.dart [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+@GenerateMocks([McndApi, Clock])
+void main() {
+  test('test fetch timetable at the end of the month', () async {
+    final api = MockMcndApi();
+    const mapper = Mapper();
+    final clock = MockClock();
+    when(clock.now()).thenReturn(DateTime(2021, 5, 25));
+
+    final may = ((await apiPrayerMonthResponse as List)[0] as List)
```
Comment (TarekkMA) [inline test/data/network/prayer_times_remote_source_test.dart:22]: Change may to month

File: test/data/network/prayer_times_remote_source_test.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    expect(prayers[5].date.day, 30);
+    expect(prayers[5].date.month, 5);
+    expect(prayers[6].date.day, 31);
+    expect(prayers[6].date.month, 5);
+    expect(prayers[7].date.day, 1);
+    expect(prayers[7].date.month, 6);
+    expect(prayers[8].date.day, 2);
+    expect(prayers[8].date.month, 6);
+    expect(prayers[9].date.day, 3);
+    expect(prayers[9].date.month, 6);
```
Comment (TarekkMA) [inline test/data/network/prayer_times_remote_source_test.dart:62]: I don't know if putting these in a for loop would be better or not

**Review by TarekkMA — APPROVED** [review]

File: test/data/network/prayer_times_remote_source_test.dart (hunk 1 above)
Comment (gemy21ce) [reply test/data/network/prayer_times_remote_source_test.dart:22]: the actual data returned is for May

File: test/data/network/prayer_times_remote_source_test.dart (hunk 2 above)
Comment (gemy21ce) [reply test/data/network/prayer_times_remote_source_test.dart:62]: will be more complicated

**Pull request merged**